src/function/scalar/list/list_lambdas.cpp

```diff
@@ -0,0 +1,417 @@
+	lists.Orrify(count, lists_data);
+	auto list_entries = (list_entry_t *)lists_data.data;
+
+	// get the lambda expression
+	auto &func_expr = (BoundFunctionExpression &)state.expr;
+	auto &info = (ListLambdaBindData &)*func_expr.bind_info;
+	auto &lambda_expr = info.lambda_expr;
+
+	// get the child vector and child data
+	auto lists_size = ListVector::GetListSize(lists);
+	auto &child_vector = ListVector::GetEntry(lists);
+	VectorData child_data;
+	child_vector.Orrify(lists_size, child_data);
+
+	// to slice the child vector
+	SelectionVector sel(STANDARD_VECTOR_SIZE);
+
+	// this vector never contains more than one element
+	vector<LogicalType> result_types;
+	result_types.push_back(lambda_expr->return_type);
+
+	// non-lambda parameter columns
+	vector<VectorData> columns;
+	vector<idx_t> indexes;
+	vector<SelectionVector> sel_vectors;
+
+	vector<LogicalType> types;
+	types.push_back(child_vector.GetType());
+
+	// skip the list column
+	for (idx_t i = 1; i < args.ColumnCount(); i++) {
+		columns.emplace_back(VectorData());
+		args.data[i].Orrify(count, columns[i - 1]);
+		indexes.push_back(0);
+		sel_vectors.emplace_back(SelectionVector(STANDARD_VECTOR_SIZE));
+		types.push_back(args.data[i].GetType());
+	}
+
+	// get the expression executor
+	ExpressionExecutor expr_executor(*lambda_expr);
+
+	// these are only for the list_filter
+	vector<idx_t> lists_len;
+	idx_t curr_list_len = 0;
+	idx_t curr_list_offset = 0;
+	idx_t appended_lists_cnt = 0;
+	idx_t curr_original_list_len = 0;
+
+	if (!IS_TRANSFORM) {
+		lists_len.reserve(count);
+	}
+
+	// loop over the child entries and create chunks to be executed by the expression executor
+	idx_t elem_cnt = 0;
+	idx_t offset = 0;
+	for (idx_t row_idx = 0; row_idx < count; row_idx++) {
+
+		auto lists_index = lists_data.sel->get_index(row_idx);
+		const auto &list_entry = list_entries[lists_index];
+
+		if (!IS_TRANSFORM) {
+			lists_len.push_back(list_entry.length);
+		}
+
+		// set the result to NULL for this row
+		if (!lists_data.validity.RowIsValid(lists_index)) {
+			result_validity.SetInvalid(row_idx);
+			continue;
+		}
+
+		// set the length and offset of the resulting lists of list_transform
+		if (IS_TRANSFORM) {
+			result_entries[row_idx].offset = offset;
+			result_entries[row_idx].length = list_entry.length;
+			offset += list_entry.length;
+		}
+
+		// empty list, nothing to execute
+		if (list_entry.length == 0) {
+			continue;
+		}
+
+		// get the data indexes
+		for (idx_t col_idx = 0; col_idx < args.ColumnCount() - 1; col_idx++) {
+			indexes[col_idx] = columns[col_idx].sel->get_index(row_idx);
+		}
+
+		// iterate list elements and create transformed expression columns
+		for (idx_t child_idx = 0; child_idx < list_entry.length; child_idx++) {
+
+			// reached STANDARD_VECTOR_SIZE elements
+			if (elem_cnt == STANDARD_VECTOR_SIZE) {
+
+				DataChunk input_chunk;
+				DataChunk lambda_chunk;
+				input_chunk.InitializeEmpty(types);
+				lambda_chunk.Initialize(result_types);
```

Comment:
There seems to be a lot of allocation/initialization happening here, can we do this once in the operator state rather than re-allocating all these chunks in every function call?